src/Orleans/Runtime/InvokableObjectManager.cs

```diff
@@ -0,0 +1,280 @@
+                return Task.CompletedTask;
+
+            object deepCopy = null;
+            try
+            {
+                // we're expected to notify the caller if the deep copy failed.
+                deepCopy = this.serializationManager.DeepCopy(resultObject);
+            }
+            catch (Exception exc2)
+            {
+                this.runtimeClient.SendResponse(message, Response.ExceptionResponse(exc2));
+                this.logger.Warn(
+                    ErrorCode.ProxyClient_OGC_SendResponseFailed,
+                    "Exception trying to send a response.",
+                    exc2);
+                return Task.CompletedTask;
+            }
+
+            // the deep-copy succeeded.
+            this.runtimeClient.SendResponse(message, new Response(deepCopy));
+            return Task.CompletedTask;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        private void ReportException(Message message, Exception exception)
+        {
+            var request = (InvokeMethodRequest)message.GetDeserializedBody(this.serializationManager);
+            switch (message.Direction)
+            {
+                default:
+                    throw new InvalidOperationException();
+                case Message.Directions.OneWay:
+                {
+                    this.logger.Error(
+                        ErrorCode.ProxyClient_OGC_UnhandledExceptionInOneWayInvoke,
+                        String.Format(
+                            "Exception during invocation of notification method {0}, interface {1}. Ignoring exception because this is a one way request.",
+                            request.MethodId,
+                            request.InterfaceId),
+                        exception);
+                    break;
+                }
+                case Message.Directions.Request:
+                {
+                    Exception deepCopy = null;
+                    try
+                    {
+                        // we're expected to notify the caller if the deep copy failed.
+                        deepCopy = (Exception)this.serializationManager.DeepCopy(exception);
+                    }
+                    catch (Exception ex2)
+                    {
+                        this.runtimeClient.SendResponse(message, Response.ExceptionResponse(ex2));
+                        this.logger.Warn(
+                            ErrorCode.ProxyClient_OGC_SendExceptionResponseFailed,
+                            "Exception trying to send an exception response", ex2);
+                        return;
+                    }
+                    // the deep-copy succeeded.
+                    var response = Response.ExceptionResponse(deepCopy);
+                    this.runtimeClient.SendResponse(message, response);
+                    break;
+                }
+            }
+        }
+
+        public class LocalObjectData
```

Comment:
Is there a reason why it is made public, since everything is internal, it has Dispose, but not IDisposable.

Reply:
No reason, I should change it. The Dispose should be moved above this class, it's just an illusion